### PR TITLE
Back button fix and refactor

### LINKED ISSUE #2322: eCR Library back button fixes
## Description

There are 2 issues with the back button:
* It's a universal browser back button, not a universal link back to the eCR Library
* It's not sticky to the top of the page

## Steps to reproduce

_List out the steps in detail to reproduce the bug_

Issue 1 repro option 1:
1. Open the non integrated viewer homepage
2. Click on an eCR
3. Use the side nav to click on a sub-section
4. Scroll up and click on the back button
5. Directed back to the sub-section of the eCR, not back to the Library

Issue 1 repro option 2:
1. open the non integrated viewer homepage
2. have links to 2 different eCRs
3. open one eCR 
4. then paste a new eCR url into the same browser window
5. click back
6. observe you go back to the first eCR you viewed
7. click back again
8. now you can get back to the Library

Issue 2 repro:
1. Open the eCR Library
2. Open an eCR
3. Scroll down 
4. Notice that you can't see the Back Button anymore

## Expected Result

Back to eCR Library button should always go back to the eCR Library.
Back button should always be sticky to the top of the page.

## Actual result

see video:

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBelE0Qmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--367faedfc0f5aa0b669905b5d832db821619ddd8/Screen%20Recording%202024-08-05%20at%206.34.21%E2%80%AFPM.mov)

## Priority

_Provide an estimate for level of impact the bug has today by placing an X in the applicable bracket_

#### Priority
- [ ] **Low priority** _The bug does not have to be fixed immediately. High and medium priority bugs should be addressed first._
- [x] **Medium priority** _Typically, medium-priority bugs do not affect customers directly and therefore can be fixed in the normal course of testing and development._
- [ ] **High priority** _High priority bugs must be addressed immediately. They often affect the app’s functionality and impact customers and their user experience. They must take priority._

### PR DESCRIPTION
**Pull request by gordonfarrell**
# PULL REQUEST

## Summary
Updates back button to always return to the eCR Library page and made it sticky. Also refactored the button into its own component and used that component for both the viewer and loading page. Tweaked a couple other bits of the loading styling to make things a bit less jumpy

## Related Issue
Fixes #2322

## Acceptance Criteria

- [x] Back to eCR Library button should always go back to the eCR Library.
- [ ] Back button should always be sticky to the top of the page.

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx [hunk 1]
```diff
@@ -217,13 +218,14 @@ const SideNav: React.FC = () => {
...
   return (
-    <div className="nav-wrapper padding-top-1-25">
+    <div className="nav-wrapper">
       <nav
         className={classNames("sticky-nav", {
           "top-0": !isNonIntegratedViewer,
           "top-550": isNonIntegratedViewer,
         })}
       >
+        <BackButton />
```
Comment (BobanL) [inline containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:228]: So now when it is non integrated there is a slight gap between the start of the side nav and the start of the main content 
Now:
image

Before:
image

Comment (BobanL) [reply containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:228]: Also, it seems like that gap also stays when scrolling down
Now:
image
Before:
image

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx
Comment (BobanL) [inline containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:1]: Just for general conversation, maybe another ticket is needed for this work. 

How do we want the side nav to react on smaller screens? Right now on my monitor I don't see the whole side nav and I don't see it until the end of the main container page.
Example: 
image
image

File: containers/ecr-viewer/src/app/view-data/components/BackButton.tsx
Comment (BobanL) [inline containers/ecr-viewer/src/app/view-data/components/BackButton.tsx:1]: Probably outside the scope of this ticket, but if a user was previously on page 5, this change will send them straight to page 1 again. Is that OK behavior?

**Review by BobanL — COMMENTED** [review]
Left a comment about some spacing when it's non integrated and when scrolling and then another couple general questions that are probably outside the scope of this ticket.

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx
Comment (gordonfarrell) [reply containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:1]: Yeah thats definitely an issue but I agree it's probably out of scope for this ticket

File: containers/ecr-viewer/src/app/view-data/components/SideNav.tsx (hunk 1 above)
Comment (gordonfarrell) [reply containers/ecr-viewer/src/app/view-data/components/SideNav.tsx:228]: This should be fixed now!

File: containers/ecr-viewer/src/app/view-data/components/BackButton.tsx
Comment (gordonfarrell) [reply containers/ecr-viewer/src/app/view-data/components/BackButton.tsx:1]: I think its fine for now as we have a ticket to eventually implement cookie storage for pagination settings

**Review by BobanL — COMMENTED** [review]
The main page looks great, but the back button on loading screen looks a bit off now.

image

**Pull request merged**